Make Cancel error compatible with anyhow

Make Cancel error compatible with anyhow: https://crates.io/crates/anyhow

```Rust
::anyhow::anyhow!(::cancellation_token::Canceled)
```

```Rust
error[E0599]: the method `anyhow_kind` exists for reference `&Canceled`, but its trait bounds were not satisfied
  --> dest/src/utils/mod.rs:28:5
   |
28 |     ::anyhow::anyhow!(::cancellation_token::Canceled)
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method cannot be called on `&Canceled` due to unsatisfied trait bounds
  --> .cargo/registry/src/index.crates.io-6f17d22bba15001f/cancellation-token-0.1.0/src/lib.rs:434:1
   |
   = note: doesn't satisfy `Canceled: Into<anyhow::Error>`
   |
   = note: doesn't satisfy `Canceled: anyhow::kind::TraitKind`
   |
   = note: doesn't satisfy `Canceled: std::fmt::Display`
   |
   = note: the following trait bounds were not satisfied:
           `Canceled: Into<anyhow::Error>`
           which is required by `Canceled: anyhow::kind::TraitKind`
           `Canceled: std::fmt::Display`
           which is required by `&Canceled: anyhow::kind::AdhocKind`
           `&Canceled: Into<anyhow::Error>`
           which is required by `&Canceled: anyhow::kind::TraitKind`
   = note: this error originates in the macro `::anyhow::anyhow` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0599`.```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,5 +1,5 @@
 use std::{
-    fmt,
+    error, fmt,
     future::{pending, Future},
     pin::{pin, Pin},
     sync::{Arc, Mutex, Weak},
@@ -434,6 +434,14 @@ pub type MayBeCanceled<T = ()> = Result<T, Canceled>;
 /// A value indicating that it has been canceled.
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Hash, PartialOrd, Ord)]
 pub struct Canceled;
+
+impl fmt::Display for Canceled {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        "operation has been cancelled".fmt(f)
+    }
+}
+
+impl error::Error for Canceled {}
 
 /// An object for which a cancellation notification is sent when dropped.
 ///

